cannon/mipsevm/versions/state.go

```diff
@@ -97,27 +79,7 @@ func (s *VersionedState) Deserialize(in io.Reader) error {
 		return err
 	}
 
-	if IsSupportedSingleThreaded(s.Version) {
-		if !arch.IsMips32 {
-			return ErrUnsupportedMipsArch
-		}
-		state := &singlethreaded.State{}
-		if err := state.Deserialize(in); err != nil {
-			return err
-		}
-		s.FPVMState = state
-		return nil
-	} else if IsSupportedMultiThreaded(s.Version) {
-		if !arch.IsMips32 {
-			return ErrUnsupportedMipsArch
-		}
-		state := &multithreaded.State{}
-		if err := state.Deserialize(in); err != nil {
-			return err
-		}
-		s.FPVMState = state
-		return nil
-	} else if IsSupportedMultiThreaded64(s.Version) {
+	if IsSupportedMultiThreaded64(s.Version) {
```

Comment:
I think we can also delete [MarshalJSON](https://github.com/ethereum-optimism/optimism/blob/2178914388657d86bf7f645597876d1fa1c9d861/cannon/mipsevm/versions/state.go#L97-L107) in this file.

Reply:
We need to keep to to prevent JSON marshalling but have simplified to just always return an error since there are no supported states that allow JSON marshalling.